openpype/tools/settings/settings/search.py

```diff
@@ -0,0 +1,137 @@
+from functools import partial
+import re
+
+from Qt import QtCore, QtWidgets
+from openpype.tools.utils.models import TreeModel, Item
+from openpype.tools.utils.lib import schedule
+
+
+def get_entity_children(entity):
+
+    if hasattr(entity, "values"):
+        return entity.values()
+    return []
+
+
+class RecursiveSortFilterProxyModel(QtCore.QSortFilterProxyModel):
```

Comment:
`RecursiveSortFilterProxyModel` can be imported from `openpype.tools.utils`.

Reply:
That's correct. But - this is a changed one. This filters to ALL columns and actually uses Qt 5.10+ feature to enable recursive filtering that Qt allows. Since I assumed the Settings tool would never be used in older Qt (e.g. in hosts with older Python, etc.)

In essence this class name should just be updated to match what it does instead - but I couldn't think of a good name. Maybe `FilterAllColumnsProxyModel`?